app/controllers/oai_controller.rb

```diff
@@ -0,0 +1,44 @@
+class OaiController < ApplicationController
+  Record = Struct.new(:id, :date, :pbcore)
+  ROWS = 100
+
+  def index
+    @verb = params.delete(:verb)
+    fail("Unsupported verb: #{@verb}") unless @verb == 'ListRecords'
+
+    @metadata_prefix = params.delete(:metadataPrefix) || 'pbcore'
+    fail("Unsupported metadataPrefix: #{@metadata_prefix}") unless @metadata_prefix == 'pbcore'
+
+    resumption_token = params.delete(:resumptionToken) || '0'
+    fail("Unsupported resumptionToken: #{resumption_token}") unless resumption_token =~ /^\d*$/
+    start = resumption_token.to_i
+
+    unsupported = params.keys - %w(action controller format)
+    fail("Unsupported params: #{unsupported}") unless unsupported.empty?
+
+    @response_date = Time.now.strftime('%FT%T')
+
+    @records =
+      RSolr.connect(url: 'http://localhost:8983/solr/')
+      .get('select', params: {
+             'q' => '*:*',
+             'fl' => 'id,timestamp,xml',
+             'rows' => ROWS,
+             'start' => start
+           })['response']['docs'].map do |d|
+        Record.new(
+          d['id'],
+          d['timestamp'],
+          d['xml'].gsub('<?xml version="1.0" encoding="UTF-8"?>', '').strip)
+      end
+
+    # Not ideal: they'll need to go past the end.
+    @next_resumption_token = start + ROWS unless @records.empty?
```

Comment:
@mccalluc if you retrieved `ROWS + 1` records using `'rows' => ROWS + 1` in your solr query, then you would know if there is an additional page or not.

```ruby
# if there is at least one more record, then set the resumption token
@next_resumption_token = start + ROWS if @records.count == ROWS + 1

# pop the last record off the array before sending to the view.
@records.pop
```